ivy/functional/ivy/experimental/manipulation.py

```diff
@@ -1742,7 +1775,7 @@ def put_along_axis(
     axis: int,
     /,
     *,
-    mode: str = "raise",
+    mode: Optional[str] = "assign",
```

Comment:
`Optional` should only be used when the default value is `None`. Same goes for the array and container method definitions above.